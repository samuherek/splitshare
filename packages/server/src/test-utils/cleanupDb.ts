import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
