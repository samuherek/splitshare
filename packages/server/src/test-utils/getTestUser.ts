import { Connection } from 'typeorm';
import { User } from '../features/user/entity';

export const testUser = {
  email: 'user@email.com',
};

export async function getTestUser(conn: Connection) {
  const user = await conn.getRepository(User).findOne(testUser);

  if (!user) {
    throw new Error('No Test user found');
  }

  return user;
}
