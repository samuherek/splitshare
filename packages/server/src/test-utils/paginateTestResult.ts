import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
