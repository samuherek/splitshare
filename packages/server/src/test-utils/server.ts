import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { getSchema } from '../schema';

export default async function createTestServer(ctx: any) {
  const schema = await getSchema();

  const server = new ApolloServer({
    schema,
    mockEntireSchema: false,
    mocks: true,
    context: () => ctx,
  });

  return createTestClient(server);
}
