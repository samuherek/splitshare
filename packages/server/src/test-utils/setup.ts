// @ts-ignore
import * as tryToCatch from 'try-to-catch';
import { Connection } from 'typeorm';
import { connectTestDatabase } from './testConnection';

(async () => {
  const [error, conn]: [Error, Connection] = await tryToCatch(
    connectTestDatabase,
    { drop: true }
  );

  if (error) {
    throw new Error(error.message);
  }
})();
