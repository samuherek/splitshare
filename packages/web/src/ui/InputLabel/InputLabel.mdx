---
name: InputLabel
---

import { Playground, Props } from 'docz';
import InputLabel from './InputLabel';

# InputLabel

Input label components are used in the `<TextField />` component. It's specifically to support the the input component

### Props

<Props of={InputLabel} />

### Playground

<Playground>
  <InputLabel />
</Playground>
