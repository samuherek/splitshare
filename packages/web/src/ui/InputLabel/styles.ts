import styled from 'styled-components';

export const LabelStandard = styled.label`
  display: block;
  margin-bottom: 5px;

  &.focused {
  }

  &.disabled {
    pointer-events: none;
  }
`;
