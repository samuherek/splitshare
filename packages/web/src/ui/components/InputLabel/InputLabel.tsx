import clsx from 'clsx';
import React from 'react';
import FormLabel from '../FormLabel';
import { LabelStandard } from './styles';

interface Props {
  className?: string;
  variant?: 'standard';
}

export const classes = {
  root: 'SSInputLabel',
};

const variantComponent = {
  standard: LabelStandard,
};

const InputLabel = React.forwardRef<Props, any>(function InputLabel(
  props,
  ref
) {
  const { className, variant = 'standard', ...rest } = props;

  // @ts-ignore
  const LabelComponent = variantComponent[variant];

  return (
    <FormLabel
      className={clsx(className, classes.root)}
      as={LabelComponent}
      ref={ref}
      {...rest}
    />
  );
});

export default InputLabel;
