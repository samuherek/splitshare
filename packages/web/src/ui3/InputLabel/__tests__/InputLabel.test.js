import React from 'react';
import { render } from '../../../test-utils/test-utils';
import InputLabel, { classes } from '../InputLabel';

test('should match snapshot', () => {
  const { getByText } = render(<InputLabel>Label</InputLabel>);
  expect(getByText(/label/i)).toMatchSnapshot();
});

test('should render <InputLabel /> correctly', () => {
  const { getByText } = render(<InputLabel>Label</InputLabel>);
  const label = getByText(/label/i);

  expect(label).toHaveProperty('nodeName', 'LABEL');
  expect(label).toHaveClass(classes.root);
  expect(label).toHaveStyleRule('color', expect.any(String));
  // TODO: look why this is not working
  // expect(label).toHaveStyleRule('pointer-events', 'none', {
  //   modifier: '.disabled',
  // });
});
