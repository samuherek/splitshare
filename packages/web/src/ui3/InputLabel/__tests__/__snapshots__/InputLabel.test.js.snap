// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  line-height: 1;
  padding: 0;
  font-size: 1em;
  color: inherit;
}

.c1 {
  display: block;
  margin-bottom: 5px;
  color: #a0a2ad;
}

.c1.focused {
  color: #131314;
}

.c1.disabled {
  pointer-events: none;
}

<label
  class="SSInputLabel SPFormLabel c0 c1"
>
  Label
  <span
    class="SSFromLabelOptional"
  >
    â€‰
    (Optional)
  </span>
</label>
`;
